tests/integration_tests/security/test_vulnerabilities.py

```diff
@@ -400,88 +181,71 @@ def check_vulnerabilities_files_on_guest(microvm):
-
-
-def test_vulnerabilities_files_on_guest_with_template(build_microvm_with_template):
-    """
-    Test vulnerabilities files on guest with CPU template.
-    """
-    check_vulnerabilities_files_ab(build_microvm_with_template)
 
-
-def test_vulnerabilities_files_on_guest_with_custom_template(
-    build_microvm_with_custom_template,
-):
-    """
-    Test vulnerabilities files on guest with a custom CPU template.
-    """
-    check_vulnerabilities_files_ab(build_microvm_with_custom_template)
+@pytest.fixture
+def microvm_factory_a(record_property):
+    """MicroVMFactory using revision A binaries"""
+    revision_a = global_props.buildkite_revision_a
+    bin_dir = git_clone(Path("../build") / revision_a, revision_a).resolve()
+    fc_bin = bin_dir / "firecracker"
+    jailer_bin = bin_dir / "jailer"
+    record_property("firecracker_bin", str(fc_bin))
+    uvm_factory = MicroVMFactory(fc_bin, jailer_bin)
+    yield uvm_factory
+    uvm_factory.kill()
 
 
-def test_vulnerabilities_files_on_restored_guest_with_template(
-    build_microvm_with_template, microvm_factory
-):
-    """
-    Test vulnerabilities files on a restored guest with a CPU template.
-    """
-    check_vulnerabilities_files_ab(
-        with_restore(build_microvm_with_template, microvm_factory)
-    )
+@pytest.fixture
+def uvm_any_a(microvm_factory_a, uvm_ctor, guest_kernel, rootfs, cpu_template_any):
+    """Return uvm with revision A firecracker"""
+    return uvm_ctor(microvm_factory_a, guest_kernel, rootfs, cpu_template_any)
+
+
+def test_check_vulnerability_files_ab(request, uvm_any):
+    """Test vulnerability files on guests"""
+    res_b = check_vulnerabilities_files_on_guest(uvm_any)
+    if global_props.buildkite_pr:
+        # we only get the uvm_any_a fixtures if we need it
+        uvm_a = request.getfixturevalue("uvm_any_a")
+        res_a = check_vulnerabilities_files_on_guest(uvm_a)
+        assert res_b <= res_a
+    else:
+        assert not [x for x in res_b if "Vulnerable" in x["stdout"]]
 
 
-def test_vulnerabilities_files_on_restored_guest_with_custom_template(
-    build_microvm_with_custom_template, microvm_factory
+def test_spectre_meltdown_checker_on_guest(
+    request,
+    uvm_any,
+    spectre_meltdown_checker,
 ):
-    """
-    Test vulnerabilities files on a restored guest with a custom CPU template.
-    """
-    check_vulnerabilities_files_ab(
-        with_restore(build_microvm_with_custom_template, microvm_factory)
-    )
+    """Test with the spectre / meltdown checker on any supported guest."""
+    res_b = spectre_meltdown_checker.get_report_for_guest(uvm_any)
+    if global_props.buildkite_pr:
+        # we only get the uvm_any_a fixtures if we need it
+        uvm_a = request.getfixturevalue("uvm_any_a")
+        res_a = spectre_meltdown_checker.get_report_for_guest(uvm_a)
+        assert res_b <= res_a
+    else:
+        assert res_b == spectre_meltdown_checker.expected_vulnerabilities(
+            uvm_any.cpu_template
```

Comment:
If the template is a static one, `cpu_template` is None, so should pass `cpu_template_name` here.
```suggestion
            uvm_any.cpu_template_name
```

Reply:
Good catch! Fixed